modules/cuda/src/util.ts

```diff
@@ -59,13 +64,38 @@ export const isArrayBufferView = ArrayBuffer.isView;
 export const isIteratorResult = <T = any>(x: any):
   x is IteratorResult<T>      => { return isObject(x) && ('done' in x) && ('value' in x);};
 
-export function clampSliceArgs(len: number, lhs = 0, rhs = len): [number, number] {
-  // Adjust args similar to Array.prototype.slice. Normalize begin/end to
-  // clamp between 0 and length, and wrap around on negative indices, e.g.
-  // slice(-1, 5) or slice(5, -1)
-  // wrap around on negative start/end positions
+/**
+ * @summary Clamp begin and end ranges similar to `Array.prototype.slice`.
+ * @description Normalizes begin/end to between 0 and length, and wrap around on negative indices.
+ * @example
+ * ```typescript
+ * import {clampRange} from '@nvidia/cuda';
+ *
+ * clampRange(5)        // [0, 5]
+ * clampRange(5, 0, -1) // [0, 4]
+ * clampRange(5, -1)    // [4, 5]
+ * clampRange(5, -1, 0) // [4, 4]
+ *
+ * const ary = Array.from({length: 5}, (_, i) => i);
+ * // [0, 1, 2, 3, 4]
+ * assert(ary.slice() == ary.slice(...clampRange(ary.length)))
+ * // [0, 1, 2, 3]
```

Comment:
why wouldn't this be `[0,1,2,3,4]` if `clampRange(5)` is `[0, 5]` (above)

Reply:
Oh this is my bad. The comments are above the line that produces them. I'll change that so there's no confusion.

Reply:
Done :+1: